Providing a generic framework for Kerntypes information,
used by lcrash to resolve kernel types.

			  Hannes Reinecke <hare@suse.de>

diff -u --recursive --new-file linux-2.6.5/arch/s390/boot/Makefile linux-2.6.5-lkcd/arch/s390/boot/Makefile
--- linux-2.6.5/arch/s390/boot/Makefile	2004-04-20 14:51:20.000000000 +0200
+++ linux-2.6.5-lkcd/arch/s390/boot/Makefile	2004-04-20 14:41:44.000000000 +0200
@@ -8,7 +8,7 @@
 
 EXTRA_CFLAGS  := -DCOMPILE_VERSION=$(COMPILE_VERSION) -gstabs -I.
 
-targets := image kerntypes.o
+targets := image
 
 $(obj)/image: vmlinux FORCE
 	$(call if_changed,objcopy)
diff -u --recursive --new-file linux-2.6.5/arch/s390/boot/kerntypes.c linux-2.6.5-lkcd/arch/s390/boot/kerntypes.c
--- linux-2.6.5/arch/s390/boot/kerntypes.c	2004-04-20 14:51:18.000000000 +0200
+++ linux-2.6.5-lkcd/arch/s390/boot/kerntypes.c	1970-01-01 01:00:00.000000000 +0100
@@ -1,58 +0,0 @@
-/*
- * kerntypes.c
- *
- * Dummy module that includes headers for all kernel types of interest.
- * The kernel type information is used by the lcrash utility when
- * analyzing system crash dumps or the live system. Using the type
- * information for the running system, rather than kernel header files,
- * makes for a more flexible and robust analysis tool.
- *
- * This source code is released under the GNU GPL.
- */
-
-/* generate version for this file */
-typedef char *COMPILE_VERSION;
-
-/* General linux types */
-
-#include <linux/autoconf.h>
-#include <linux/compile.h>
-#include <linux/config.h>
-#include <linux/utsname.h>
-#include <linux/module.h>
-#include <linux/sched.h>
-#include <linux/mm.h>
-
-#include <asm/lowcore.h>
-#include <asm/debug.h>
-#include <asm/ccwdev.h>
-#include <asm/ccwgroup.h>
-#include <asm/qdio.h>
-
-/* channel subsystem driver */
-#include "drivers/s390/cio/cio.h"
-#include "drivers/s390/cio/chsc.h"
-#include "drivers/s390/cio/css.h"
-#include "drivers/s390/cio/device.h"
-#include "drivers/s390/cio/qdio.h"
-
-/* dasd device driver */
-#include "drivers/s390/block/dasd_int.h"
-#include "drivers/s390/block/dasd_diag.h"
-#include "drivers/s390/block/dasd_eckd.h"
-#include "drivers/s390/block/dasd_fba.h"
-
-/* networking drivers */
-#include "drivers/s390/net/fsm.h"
-#include "drivers/s390/net/iucv.h"
-#include "drivers/s390/net/lcs.h"
-
-/* zfcp device driver */
-#include "drivers/s390/scsi/zfcp_def.h"
-#include "drivers/s390/scsi/zfcp_fsf.h"
-
-/* include sched.c for types: 
- *    - struct prio_array 
- *    - struct runqueue
- */
-#include "kernel/sched.c"
diff -u --recursive --new-file linux-2.6.5/include/asm-alpha/kerntypes.h linux-2.6.5-lkcd/include/asm-alpha/kerntypes.h
--- linux-2.6.5/include/asm-alpha/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-alpha/kerntypes.h	2004-04-20 14:08:30.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-alpha/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* Alpha-specific header files */
+#ifndef _ALPHA_KERNTYPES_H
+#define _ALPHA_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _ALPHA_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-arm/kerntypes.h linux-2.6.5-lkcd/include/asm-arm/kerntypes.h
--- linux-2.6.5/include/asm-arm/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-arm/kerntypes.h	2004-04-20 14:23:08.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-arm/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* ARM-specific header files */
+#ifndef _ARM_KERNTYPES_H
+#define _ARM_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _ARM_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-arm26/kerntypes.h linux-2.6.5-lkcd/include/asm-arm26/kerntypes.h
--- linux-2.6.5/include/asm-arm26/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-arm26/kerntypes.h	2004-04-20 14:24:06.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-arm26/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* ARM26-specific header files */
+#ifndef _ARM26_KERNTYPES_H
+#define _ARM26_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _ARM26_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-cris/kerntypes.h linux-2.6.5-lkcd/include/asm-cris/kerntypes.h
--- linux-2.6.5/include/asm-cris/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-cris/kerntypes.h	2004-04-20 14:24:52.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-cris/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* CRIS-specific header files */
+#ifndef _CRIS_KERNTYPES_H
+#define _CRIS_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _CRIS_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-generic/kerntypes.h linux-2.6.5-lkcd/include/asm-generic/kerntypes.h
--- linux-2.6.5/include/asm-generic/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-generic/kerntypes.h	2004-04-20 14:09:13.000000000 +0200
@@ -0,0 +1,20 @@
+/*
+ * asm-generic/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* Arch-independent header files */
+#ifndef _GENERIC_KERNTYPES_H
+#define _GENERIC_KERNTYPES_H
+
+#include <linux/pci.h>
+
+#endif /* _GENERIC_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-h8300/kerntypes.h linux-2.6.5-lkcd/include/asm-h8300/kerntypes.h
--- linux-2.6.5/include/asm-h8300/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-h8300/kerntypes.h	2004-04-20 14:25:33.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-h8300/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* H8300-specific header files */
+#ifndef _H8300_KERNTYPES_H
+#define _H8300_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _H8300_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-i386/kerntypes.h linux-2.6.5-lkcd/include/asm-i386/kerntypes.h
--- linux-2.6.5/include/asm-i386/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-i386/kerntypes.h	2004-04-20 14:26:16.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-i386/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* ix86-specific header files */
+#ifndef _I386_KERNTYPES_H
+#define _I386_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _I386_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-ia64/kerntypes.h linux-2.6.5-lkcd/include/asm-ia64/kerntypes.h
--- linux-2.6.5/include/asm-ia64/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-ia64/kerntypes.h	2004-04-20 14:26:48.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-ia64/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* IA64-specific header files */
+#ifndef _IA64_KERNTYPES_H
+#define _IA64_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _IA64_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-m68k/kerntypes.h linux-2.6.5-lkcd/include/asm-m68k/kerntypes.h
--- linux-2.6.5/include/asm-m68k/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-m68k/kerntypes.h	2004-04-20 14:27:34.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-m68k/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* m68k-specific header files */
+#ifndef _M68K_KERNTYPES_H
+#define _M68K_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _M68K_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-m68knommu/kerntypes.h linux-2.6.5-lkcd/include/asm-m68knommu/kerntypes.h
--- linux-2.6.5/include/asm-m68knommu/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-m68knommu/kerntypes.h	2004-04-20 14:28:31.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-m68knommu/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* m68k/no-MMU-specific header files */
+#ifndef _M68KNOMMU_KERNTYPES_H
+#define _M68KNOMMU_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _M68KNOMMU_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-mips/kerntypes.h linux-2.6.5-lkcd/include/asm-mips/kerntypes.h
--- linux-2.6.5/include/asm-mips/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-mips/kerntypes.h	2004-04-20 14:29:03.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-mips/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* MIPS-specific header files */
+#ifndef _MIPS_KERNTYPES_H
+#define _MIPS_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _MIPS_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-mips64/kerntypes.h linux-2.6.5-lkcd/include/asm-mips64/kerntypes.h
--- linux-2.6.5/include/asm-mips64/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-mips64/kerntypes.h	2004-04-20 14:29:35.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-mips64/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* MIPS64-specific header files */
+#ifndef _MIPS64_KERNTYPES_H
+#define _MIPS64_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _MIPS64_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-parisc/kerntypes.h linux-2.6.5-lkcd/include/asm-parisc/kerntypes.h
--- linux-2.6.5/include/asm-parisc/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-parisc/kerntypes.h	2004-04-20 14:30:18.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-parisc/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* PA-RISC-specific header files */
+#ifndef _PARISC_KERNTYPES_H
+#define _PARISC_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _PARISC_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-ppc/kerntypes.h linux-2.6.5-lkcd/include/asm-ppc/kerntypes.h
--- linux-2.6.5/include/asm-ppc/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-ppc/kerntypes.h	2004-04-20 14:30:47.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-ppc/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* PowerPC-specific header files */
+#ifndef _PPC_KERNTYPES_H
+#define _PPC_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _PPC_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-ppc64/kerntypes.h linux-2.6.5-lkcd/include/asm-ppc64/kerntypes.h
--- linux-2.6.5/include/asm-ppc64/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-ppc64/kerntypes.h	2004-04-20 14:31:15.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-ppc64/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* PPC64-specific header files */
+#ifndef _PPC64_KERNTYPES_H
+#define _PPC64_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _PPC64_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-s390/kerntypes.h linux-2.6.5-lkcd/include/asm-s390/kerntypes.h
--- linux-2.6.5/include/asm-s390/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-s390/kerntypes.h	2004-04-20 14:10:04.000000000 +0200
@@ -0,0 +1,46 @@
+/*
+ * asm-s390/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* S/390 specific header files */
+#ifndef _S390_KERNTYPES_H
+#define _S390_KERNTYPES_H
+
+#include <asm/lowcore.h>
+#include <asm/debug.h>
+#include <asm/ccwdev.h>
+#include <asm/ccwgroup.h>
+#include <asm/qdio.h>
+
+/* channel subsystem driver */
+#include "../../drivers/s390/cio/cio.h"
+#include "../../drivers/s390/cio/chsc.h"
+#include "../../drivers/s390/cio/css.h"
+#include "../../drivers/s390/cio/device.h"
+#include "../../drivers/s390/cio/qdio.h"
+
+/* dasd device driver */
+#include "../../drivers/s390/block/dasd_int.h"
+#include "../../drivers/s390/block/dasd_diag.h"
+#include "../../drivers/s390/block/dasd_eckd.h"
+#include "../../drivers/s390/block/dasd_fba.h"
+
+/* networking drivers */
+#include "../../drivers/s390/net/fsm.h"
+#include "../../drivers/s390/net/iucv.h"
+#include "../../drivers/s390/net/lcs.h"
+
+/* zfcp device driver */
+#include "../../drivers/s390/scsi/zfcp_def.h"
+#include "../../drivers/s390/scsi/zfcp_fsf.h"
+
+#endif /* _S390_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-sh/kerntypes.h linux-2.6.5-lkcd/include/asm-sh/kerntypes.h
--- linux-2.6.5/include/asm-sh/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-sh/kerntypes.h	2004-04-20 14:31:45.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-sh/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* Super-H-specific header files */
+#ifndef _SH_KERNTYPES_H
+#define _SH_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _SH_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-sparc/kerntypes.h linux-2.6.5-lkcd/include/asm-sparc/kerntypes.h
--- linux-2.6.5/include/asm-sparc/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-sparc/kerntypes.h	2004-04-20 14:32:13.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-sparc/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* SPARC-specific header files */
+#ifndef _SPARC_KERNTYPES_H
+#define _SPARC_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _SPARC_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-sparc64/kerntypes.h linux-2.6.5-lkcd/include/asm-sparc64/kerntypes.h
--- linux-2.6.5/include/asm-sparc64/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-sparc64/kerntypes.h	2004-04-20 14:32:52.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-sparc64/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* SPARC64-specific header files */
+#ifndef _SPARC64_KERNTYPES_H
+#define _SPARC64_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _SPARC64_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-um/kerntypes.h linux-2.6.5-lkcd/include/asm-um/kerntypes.h
--- linux-2.6.5/include/asm-um/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-um/kerntypes.h	2004-04-20 14:33:20.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-um/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* Usermode-Linux-specific header files */
+#ifndef _UM_KERNTYPES_H
+#define _UM_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _UM_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-v850/kerntypes.h linux-2.6.5-lkcd/include/asm-v850/kerntypes.h
--- linux-2.6.5/include/asm-v850/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-v850/kerntypes.h	2004-04-20 14:34:09.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-v850/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* V850-specific header files */
+#ifndef _V850_KERNTYPES_H
+#define _V850_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _V850_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/include/asm-x86_64/kerntypes.h linux-2.6.5-lkcd/include/asm-x86_64/kerntypes.h
--- linux-2.6.5/include/asm-x86_64/kerntypes.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.5-lkcd/include/asm-x86_64/kerntypes.h	2004-04-20 14:34:58.000000000 +0200
@@ -0,0 +1,21 @@
+/*
+ * asm-x86_64/kerntypes.h
+ *
+ * Arch-dependent header file that includes headers for all arch-specific 
+ * types of interest.
+ * The kernel type information is used by the lcrash utility when
+ * analyzing system crash dumps or the live system. Using the type
+ * information for the running system, rather than kernel header files,
+ * makes for a more flexible and robust analysis tool.
+ *
+ * This source code is released under the GNU GPL.
+ */
+
+/* x86_64-specific header files */
+#ifndef _X86_64_KERNTYPES_H
+#define _X86_64_KERNTYPES_H
+
+/* Use the default */
+#include <asm-generic/kerntypes.h>
+
+#endif /* _X86_64_KERNTYPES_H */
diff -u --recursive --new-file linux-2.6.5/init/kerntypes.c linux-2.6.5-lkcd/init/kerntypes.c
--- linux-2.6.5/init/kerntypes.c	2004-04-20 14:51:26.000000000 +0200
+++ linux-2.6.5-lkcd/init/kerntypes.c	2004-04-20 13:53:56.000000000 +0200
@@ -22,6 +22,8 @@
 #include <linux/kernel_stat.h>
 #include <linux/dump.h>
 
+#include <asm/kerntypes.h>
+
 #ifdef LINUX_COMPILE_VERSION_ID_TYPE
 /* Define version type for version validation of dump and kerntypes */
 LINUX_COMPILE_VERSION_ID_TYPE;
--- linux-2.6.5/arch/s390/Makefile	2004-05-13 11:31:21.396599388 +0200
+++ linux-2.6.5-lkcd/arch/s390/Makefile	2004-05-13 12:36:51.378640664 +0200
@@ -52,12 +52,12 @@
 
 boot		:= arch/$(ARCH)/boot
 
-all: image kerntypes.o
+all: image
 
 install: vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) $@
 
-image kerntypes.o: vmlinux
+image: vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) $(boot)/$@
 
 archclean:
